src/router.rs

```diff
@@ -1,90 +1,75 @@
-use hyper::{header, Body, Method, Request, Response, StatusCode};
+use crate::{
+    request::{Json, Request},
+    response::Response,
+};
+use bytes::Bytes;
+use futures::future::Future;
+use h2::{self, server::SendResponse};
 use serde_json::json;
-use std::collections::HashMap;
+use std::{collections::HashMap, error::Error as StdError};
 
-/// # RouteHandler
-///
-/// Any function matching the signature `(&Request<Body>) -> Response<Body>`
-/// can be used as a route handler.  
-pub trait RouteHandler: Send + Sync + 'static {
-    fn respond_to<'r>(&self, req: &'r Request<Body>) -> Response<Body>;
-}
-
-impl<F: Send + Sync + 'static> RouteHandler for F
-where
-    for<'r> F: Fn(&'r Request<Body>) -> Response<Body>,
-{
-    fn respond_to<'r>(&self, req: &'r Request<Body>) -> Response<Body> {
-        self(req)
-    }
-}
+pub type route_fn = fn(Request<Json>) -> Response;
 
-/// # Route
```

Comment:
Could you keep the docs? or do you anticipate some major changes in the future?

Reply:
Its going to be overhauled